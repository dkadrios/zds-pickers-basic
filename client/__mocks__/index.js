export inputControls from './inputControls'
export mappings from './mappings'
export midi from './midi'
export shiftGroups from './shiftGroups'
export shifter from './shifter'
export store from './store'
export user from './user'
export version from './version'
