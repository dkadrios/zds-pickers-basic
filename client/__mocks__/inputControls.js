export default [
  {
    controlId: 0,
    channel: 10,
    calibrationLow: 0,
    calibrationHigh: 127,
    curve: 0,
    polarity: 0,
    latching: 0,
    status: 11,
    value: 110,
    active: true,
    lit: true,
  },
  {
    controlId: 1,
    channel: 10,
    calibrationLow: 0,
    calibrationHigh: 127,
    curve: 0,
    polarity: 0,
    latching: 1,
    status: 11,
    value: 111,
    active: false,
    lit: false,
  },
  {
    controlId: 2,
    channel: 10,
    calibrationLow: 0,
    calibrationHigh: 127,
    curve: 0,
    polarity: 0,
    latching: 0,
    status: 9,
    value: 110,
    active: true,
    lit: false,
  },
  {
    controlId: 3,
    channel: 10,
    calibrationLow: 0,
    calibrationHigh: 127,
    curve: 0,
    polarity: 0,
    latching: 0,
    status: 9,
    value: 110,
    active: true,
    lit: true,
  },
  {
    controlId: 4,
    channel: 10,
    calibrationLow: 0,
    calibrationHigh: 127,
    curve: 0,
    polarity: 0,
    latching: 0,
    status: 9,
    value: 110,
    active: true,
    lit: true,
  },
  {
    controlId: 5,
    channel: 10,
    calibrationLow: 0,
    calibrationHigh: 127,
    curve: 0,
    polarity: 0,
    latching: 0,
    status: 9,
    value: 110,
    active: true,
    lit: false,
  },
]
