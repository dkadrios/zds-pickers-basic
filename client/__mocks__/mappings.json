{
  "dialogVisible": false,
  "userDialogVisible": false,
  "userImportDialogVisible": false,
  "channels": [
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    null,
    "BFD3",
    null,
    null,
    null,
    null,
    null,
    null
  ],
  "stockMappings": [
    "2box",
    "Addictive Drums",
    "Alesis DM Pro",
    "Alesis DM10",
    "Alesis DM5",
    "BFD2",
    "BFD3",
    "Battery 3",
    "E-MU Protean Acoustic",
    "E-MU Protean Bread n Butter",
    "E-MU Protean Studio",
    "E-MU Protean Synth",
    "EZ Drummer The Classic EZX",
    "EZDrummer",
    "EZDrummer Jazz Kit",
    "EZDrummer Latin Percussion",
    "General MIDI",
    "Infinite Instruments",
    "John Emrich - Articulations 1",
    "Kontakt 2",
    "Ocean Way Drums",
    "Pearl Redbox",
    "Roland HandSonic 15",
    "Roland TD-10",
    "Roland TD-10 Perc 1",
    "Roland TD-10 Perc 2",
    "Roland TD-10 Perc 3",
    "Roland TD-10 Perc 4",
    "Roland TD-12",
    "Roland TD-12 Perc 1 (Latin)",
    "Roland TD-12 Perc 2 (Indian)",
    "Roland TD-12 Perc 3 (African)",
    "Roland TD-12 Perc 4 (Salsa)",
    "Roland TD-12 Perc 5 (SFX)",
    "Roland TD-12 Perc 6-8 (User)",
    "Roland TD-20",
    "Roland TD-20 Perc",
    "Roland TD-3",
    "Roland TD-30",
    "Roland TD-30 Perc 1 (Latin)",
    "Roland TD-30 Perc 2 (Indian)",
    "Roland TD-30 Perc 3 (African)",
    "Roland TD-30 Perc 4 (Salsa)",
    "Roland TD-30 Perc 5 (SFX)",
    "Roland TD-30 Perc 6-8 (User)",
    "Roland TD-5",
    "Roland TD-6",
    "Roland TD-6 Perc 1",
    "Roland TD-6 Perc 10 (Special)",
    "Roland TD-6 Perc 2",
    "Roland TD-6 Perc 3 (Room)",
    "Roland TD-6 Perc 4 (Power)",
    "Roland TD-6 Perc 5 (Elec)",
    "Roland TD-6 Perc 6 (808)",
    "Roland TD-6 Perc 7 (Jazz)",
    "Roland TD-6 Perc 8 (Brush)",
    "Roland TD-6 Perc 9 (Perc)",
    "Roland TD-7",
    "Roland TD-8",
    "Roland TD-8 Perc 1",
    "Roland TD-8 Perc 10 (Special)",
    "Roland TD-8 Perc 2",
    "Roland TD-8 Perc 3 (Room)",
    "Roland TD-8 Perc 4 (Power)",
    "Roland TD-8 Perc 5 (Elec)",
    "Roland TD-8 Perc 6 (808)",
    "Roland TD-8 Perc 7 (Jazz)",
    "Roland TD-8 Perc 8 (Brush)",
    "Roland TD-8 Perc 9 (Perc)",
    "Roland TD-9",
    "Roland TD-9 Perc",
    "Simmons SD1000",
    "Simmons SD5X",
    "Simmons SD7",
    "Simmons SD7PK",
    "Simmons SD9K",
    "Sonic Reality iMap Extended",
    "Sonic Reality iMap Standard",
    "Studio ProFiles",
    "Superior Drummer",
    "Yamaha DTX 700",
    "Yamaha DTXplorer",
    "Zendrum Stompblock",
    "Zendrum TRS",
    "Zendrum TRS 2.0"
  ],
  "userMappings": ["BFD3 Custom ZenEdit", "reaper"]
}
