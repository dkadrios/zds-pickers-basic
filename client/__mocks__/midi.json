{
  "devices": [
    {
      "id": "-1325431060",
      "manufacturer": "Zendrum Studio",
      "name": "ZDS Shifter",
      "type": "output",
      "version": "",
      "state": "connected",
      "connection": "open"
    },
    {
      "id": "-312145282",
      "manufacturer": "Zendrum Studio",
      "name": "ZDS Shifter",
      "type": "input",
      "version": "",
      "state": "connected",
      "connection": "open"
    }
  ],
  "listeningDevices": ["-312145282"]
}
