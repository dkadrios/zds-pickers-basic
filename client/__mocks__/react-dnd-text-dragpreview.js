/* eslint-disable import/prefer-default-export, jsx-a11y/alt-text */
import React from 'react'

export const createDragPreview = () => <img src={null} />
