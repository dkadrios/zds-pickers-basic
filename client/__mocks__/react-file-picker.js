/* eslint-disable import/prefer-default-export */
export const FilePicker = () => <div>content</div>
