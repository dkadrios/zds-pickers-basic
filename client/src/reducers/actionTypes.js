export default {
  ADD_ENTRY: 'ADD_ENTRY',
  CANCEL_ENTRY_EDIT: 'CANCEL_ENTRY_EDIT',
  CHANGE_CONTROL_TYPE: 'CHANGE_CONTROL_TYPE',
  CHANGE_ENTRY_CHANNEL: 'CHANGE_ENTRY_CHANNEL',
  CHANGE_ENTRY_STATUS: 'CHANGE_ENTRY_STATUS',
  CHANGE_ENTRY_VALUE: 'CHANGE_ENTRY_VALUE',
  CHANGE_GROUP_CHANNEL: 'CHANGE_GROUP_CHANNEL',
  CHANGE_GROUP_VALUE: 'CHANGE_GROUP_VALUE',
  CHANGE_INPUT_CONTROL_CHANNEL: 'CHANGE_INPUT_CONTROL_CHANNEL',
  CHANGE_INPUT_CONTROL_LATCHING: 'CHANGE_INPUT_CONTROL_LATCHING',
  CHANGE_INPUT_CONTROL_POLARITY: 'CHANGE_INPUT_CONTROL_POLARITY',
  CHANGE_INPUT_CONTROL_VALUE: 'CHANGE_INPUT_CONTROL_VALUE',
  CHANGE_SELECTED_GROUP: 'CHANGE_SELECTED_GROUP',
  CHECKED_REGISTRATION: 'CHECKED_REGISTRATION',
  CLEAR_ALL_SELECTIONS: 'CLEAR_ALL_SELECTIONS',
  CONFIRM_FACTORY_RESET: 'CONFIRM_FACTORY_RESET',
  DELETE_MAPPING: 'DELETE_MAPPING',
  DEVICE_REGISTERED: 'DEVICE_REGISTERED',
  DISMISS_ERROR: 'DISMISS_ERROR',
  EDIT_ENTRY: 'EDIT_ENTRY',
  ENTRY_META_CLICKED: 'ENTRY_META_CLICKED',
  EXPORT_SETTINGS_PACKET: 'EXPORT_SETTINGS_PACKET',
  EXPORT_SETTINGS: 'EXPORT_SETTINGS',
  FACTORY_RESET: 'FACTORY_RESET',
  GET_SYSEX_CONTROLS: 'GET_SYSEX_CONTROLS',
  GET_SYSEX_GROUPS: 'GET_SYSEX_GROUPS',
  GET_SYSEX_MODEL: 'GET_SYSEX_MODEL',
  GET_SYSEX_VERSION: 'GET_SYSEX_VERSION',
  HARDWARE_TEST_FACTORY_RESET: 'HARDWARE_TEST_FACTORY_RESET',
  HARDWARE_TEST_NEXT: 'HARDWARE_TEST_NEXT',
  HARDWARE_TEST_SEND_ENTRIES: 'HARDWARE_TEST_SEND_ENTRIES',
  HARDWARE_TEST_SEND_MESSAGE: 'HARDWARE_TEST_SEND_MESSAGE',
  HARDWARE_TEST_SEND_MESSAGES: 'HARDWARE_TEST_SEND_MESSAGES',
  HIDE_HARDWARE_TEST_DIALOG: 'HIDE_HARDWARE_TEST_DIALOG',
  HIDE_REGISTRATION_DLG: 'HIDE_REGISTRATION_DLG',
  HIDE_REGISTRATION_NAG: 'HIDE_REGISTRATION_NAG',
  IMPORT_MAPPING: 'IMPORT_MAPPING',
  IMPORT_SETTINGS: 'IMPORT_SETTINGS',
  INVALID_SETTINGS_FILE_ACK: 'INVALID_SETTINGS_FILE_ACK',
  INVALID_SETTINGS_FILE: 'INVALID_SETTINGS_FILE',
  MAPPING_CHANGED: 'MAPPING_CHANGED',
  MIDI_IDLE: 'MIDI_IDLE',
  MIDI_IN_ACTIVITY: 'MIDI_IN_ACTIVITY',
  MIDI_OUT_ACTIVITY: 'MIDI_OUT_ACTIVITY',
  NOT_RESPONDING: 'NOT_RESPONDING',
  RECEIVED_AVAILABILITY: 'RECEIVED_AVAILABILITY',
  RECEIVED_CONTROLS: 'RECEIVED_CONTROLS',
  RECEIVED_ENTRY: 'RECEIVED_ENTRY',
  RECEIVED_GROUP: 'RECEIVED_GROUP',
  RECEIVED_GROUPS: 'RECEIVED_GROUPS',
  RECEIVED_MODEL: 'RECEIVED_MODEL',
  RECEIVED_STATE: 'RECEIVED_STATE',
  RECEIVED_VERSION: 'RECEIVED_VERSION',
  REMOVE_ENTRY: 'REMOVE_ENTRY',
  REPORT_ERROR: 'REPORT_ERROR',
  RESTART: 'RESTART',
  SAVE_ENTRY_EDIT: 'SAVE_ENTRY_EDIT',
  SEARCHED_FOR_SHIFTER: 'SEARCHED_FOR_SHIFTER',
  SELECT_ALL_ENTRIES: 'SELECT_ALL_ENTRIES',
  SELECT_SHIFT_ENTRY: 'SELECT_SHIFT_ENTRY',
  SET_FLAGS: 'SET_FLAGS',
  SHIFTER_FOUND: 'SHIFTER_FOUND',
  SHIFTER_MISSING: 'SHIFTER_MISSING',
  SHOW_EXPORT_DIALOG: 'SHOW_EXPORT_DIALOG',
  SHOW_HARDWARE_TEST_DIALOG: 'SHOW_HARDWARE_TEST_DIALOG',
  SHOW_IMPORT_DIALOG: 'SHOW_IMPORT_DIALOG',
  SHOW_MAPPINGS_DLG: 'SHOW_MAPPINGS_DLG',
  SHOW_REGISTRATION_DLG: 'SHOW_REGISTRATION_DLG',
  SHOW_USER_MAPPINGS_DIALOG: 'SHOW_USER_MAPPINGS_DIALOG',
  SORT_GROUP: 'SORT_GROUP',
  TEST_INTERFACE_FOUND: 'TEST_INTERFACE_FOUND',
  TEST_INTERFACE_MISSING: 'TEST_INTERFACE_MISSING',
}
