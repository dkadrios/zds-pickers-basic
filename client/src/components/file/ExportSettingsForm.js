import PropTypes from 'prop-types'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { actions as shifterActions } from 'reducers/shifter'
import { fieldFilename, fieldRequired } from 'fp/strings'

export const ExportSettingsForm = ({ error, handleSubmit }) => (
  <form onSubmit={handleSubmit}>
    {/* <Field
      autoFocus
      component={FormInput}
      label="Filename"
      name="exportFilename"
      type="text"
      validate={[fieldRequired, fieldFilename]}
    /> */}

    {Boolean(error) && <strong>{error}</strong>}
  </form>
)

ExportSettingsForm.propTypes = {
  handleSubmit: PropTypes.func.isRequired,
  error: PropTypes.string,
}
ExportSettingsForm.defaultProps = { error: '' }

export const mapStateToProps = ({ shifter }) => ({
  initialValues: { exportFilename: shifter.exportFilename },
})

export const mapDispatchToProps = dispatch => bindActionCreators(shifterActions, dispatch)
const formOptions = {
  form: 'exportSettingsForm',
  enableReinitialize: true,
  onSubmit: /* istanbul ignore next */ (values, dispatch) => dispatch(shifterActions.exportSettings(values)),
}

/* eslint-disable */
export default connect(
  mapStateToProps,
  mapDispatchToProps,
)((ExportSettingsForm))
