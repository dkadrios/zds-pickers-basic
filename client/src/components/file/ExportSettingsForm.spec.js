import set from 'lodash/fp/set'
import { shifter, store } from '../../../__mocks__'
import actionTypes from '../../reducers/actionTypes'
import { ExportSettingsForm, mapDispatchToProps, mapStateToProps } from './ExportSettingsForm'

const handleSubmit = jest.fn()

const props = {
  shifter,
  handleSubmit,
  error: '',
}

describe('ExportSettingsForm tests', () => {
  beforeEach(jest.clearAllMocks)

  it('renders correctly', () => {
    shallowExpect(ExportSettingsForm)(props).toMatchSnapshot()
  })

  it('renders errors correctly', () => {
    shallowExpect(ExportSettingsForm)(set('error', 'error message')(props)).toMatchSnapshot()
  })

  it('should utilize mapStateToProps', () => {
    expect(mapStateToProps(store)).toEqual({ initialValues: { exportFilename: shifter.exportFilename } })
  })

  it('should utilize mapDispatchToProps', () => {
    const dispatch = jest.fn()
    const exportFilename = 'filename.txt'
    mapDispatchToProps(dispatch).exportSettings({ exportFilename })
    expect(dispatch.mock.calls[0][0]).toEqual({
      exportDialogVisible: false,
      exportFilename,
      type: actionTypes.EXPORT_SETTINGS,
    })
  })
})
