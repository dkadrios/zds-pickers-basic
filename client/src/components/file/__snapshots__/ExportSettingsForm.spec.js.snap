// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExportSettingsForm tests renders correctly 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <Field
    autoFocus={true}
    component={[Function]}
    label="Filename"
    name="exportFilename"
    type="text"
    validate={
      Array [
        [Function],
        [Function],
      ]
    }
  />
</form>
`;

exports[`ExportSettingsForm tests renders errors correctly 1`] = `
<form
  onSubmit={[MockFunction]}
>
  <Field
    autoFocus={true}
    component={[Function]}
    label="Filename"
    name="exportFilename"
    type="text"
    validate={
      Array [
        [Function],
        [Function],
      ]
    }
  />
  <strong>
    error message
  </strong>
</form>
`;
